beacon-chain/sync/data_columns.go

```diff
@@ -59,6 +146,13 @@ func RequestDataColumnSidecarsByRoot(
 		return nil, errors.Wrap(err, "couldn't get admissible peers for data columns")
 	}
 
+	// If the request was non-empty but no peers were found for any needed column,
+	// return the specific error immediately.
+	if len(dataColumnsToFetch) > 0 && len(dataColumnsByAdmissiblePeer) == 0 {
```

Comment:
`len(dataColumnsToFetch) > 0` is already tested at the very beginning of the function.

Reply:
7c4c6cd33f94d584e79ef683e92325136f5a1bb3